packages/contracts/src/Market.sol

```diff
@@ -19,6 +19,7 @@ contract Market {
       uint32 amount;
       uint256 price;
       CrossAddress seller;
+      bool lock;
```

Comment:
The Order use in `buy` method now has to be `Order storage order` so that lock is preserved between external function calls.

Reply:
done